domain/bakerystorage/state/config.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/canonical/sqlair"
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery"
+	"github.com/juju/errors"
+
+	coredatabase "github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain"
+	internaldatabase "github.com/juju/juju/internal/database"
+)
+
+type ConfigState struct {
+	*domain.StateBase
+}
+
+func NewConfigState(factory coredatabase.TxnRunnerFactory) *ConfigState {
+	return &ConfigState{
+		StateBase: domain.NewStateBase(factory),
+	}
+}
+
+func (st *ConfigState) InitialiseBakeryConfig(
+	ctx context.Context,
+	localUsersKey,
+	localUsersThirdPartyKey,
+	externalUsersThirdPartyKey,
+	offersThirdPartyKey *bakery.KeyPair,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	cfg := Config{
+		LocalUsersPrivateKey:              &keyScanner{localUsersKey.Private.Key},
+		LocalUsersPublicKey:               &keyScanner{localUsersKey.Public.Key},
+		LocalUsersThirdPartyPrivateKey:    &keyScanner{localUsersThirdPartyKey.Private.Key},
+		LocalUsersThirdPartyPublicKey:     &keyScanner{localUsersThirdPartyKey.Public.Key},
+		ExternalUsersThirdPartyPrivateKey: &keyScanner{externalUsersThirdPartyKey.Private.Key},
+		ExternalUsersThirdPartyPublicKey:  &keyScanner{externalUsersThirdPartyKey.Public.Key},
+		OffersThirdPartyPrivateKey:        &keyScanner{offersThirdPartyKey.Private.Key},
+		OffersThirdPartyPublicKey:         &keyScanner{offersThirdPartyKey.Public.Key},
+	}
+
+	initialiseConfigStmt, err := sqlair.Prepare("INSERT INTO bakery_config (*) VALUES ($Config.*)", Config{})
```

Comment:
Can just use `cfg` instead of a new `Config{}` struct as an arg to prepare.

Reply:
I think I prefer using an empty `Config{}` struct. Passing in `cfg` could be misleading